server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -177,18 +185,18 @@ public ColumnPositionResolver resolve(@Nonnull ColumnPositionResolver toResolve)
             return merged;
         }
 
-        public void addUnpositionedMapper(BuilderContext context, Mapper mapper) {
+        private void addUnpositionedMapper(Mapper mapper, int depth) {
             this.unpositionedMappers.put(
-                // mappers are input in depth first order but want to convert it to
-                // breadth-first, then insertion order second. This way, parent's position < children's positions.
-                // This implementation relies on the fact that column positions are limited by Integer.MAX_VALUE.
-                // Since Integer.MAX_VALUE * Integer.MAX_VALUE + Integer.MAX_VALUE < Long.MAX_VALUE,
-                // every invocation should have a unique key.
-                (long) context.contentPath.currentDepth() * Integer.MAX_VALUE + this.unpositionedMappers.size(),
+                // mappers are input in depth first order but want to convert it to breadth first order.
+                // That way, parent's position < children's positions.
+
+                // Below calculates a unique key for each unpositioned mappers.
+                // Integer.MAX_VALUE * Integer.MAX_VALUE + Integer.MAX_VALUE < Long.MAX_VALUE 
+                (long) depth * Integer.MAX_VALUE + this.unpositionedMappers.size(),
```

Comment:
How difficult would it be to change how mappers are added to avoid having to do this?

Reply:
updated